charters/PipelinePortability/charter.md

```diff
@@ -0,0 +1,70 @@
+# [Pipeline Portability Service](mailto:pipelines-team@data.humancellatlas.org)
+
+## Description
+
+Reproducibility of analysis by others is a cornerstone of science. To satisfy this need, HCA data processing pipelines should be able to be executed by others in other projects and infrastructure. The Portability Service confirms that pipelines developed for the HCA DCP can be successfully executed outside of the HCA DCP. Confirmation that HCA data processing pipelines can execute internally in the DCP is not assessed by this service but such confirmation is a part of our standard development and testing practices.   
+
+## Objectives
+
+To confirm there are no technical barriers to executing pipelines outside of the DCP infrastructure, the Portability Service executes pipelines in a set of other environments and verifies that they run successfully. The Portability Service implementation team as well as third parties can submit environments to be included in the Portability Service's test set.   
+
+## Definitions
+
+__3rd party:__ A developer outside of the team implementing the portability service.  
+__3rd party backend:__ An execution environment or storage platform that could be used to execute data processing pipelines but is not officially used or supported by the DCP.  
+__HCA DCP:__ The Human Cell Atlas Data Coordination Platform is the data center for the HCA.  
+
+## In-scope
+
+* Programmatic authorized access to submit pipelines (to be automatically executed in all integrated backends)  
+* Programmatic authorized access to extend the backend of the Portability Service  
+* A UI to describe the success of pipeline execution on different backends  
+* Documentation to describe how to submit a backend or pipeline to the system  
+* Documentation on portability best practices that are learned as the system is used  
+* Incorporating backends explicitly used by the infrastructure of the HCA DCP (ie. testing platforms used by the data storage service, execution environments used by the data processing service, and workflows languages officially supported by the project)  
+
+## Scientific "guardrails"
+
+Portability service will not assess the scientific accuracy of pipelines, and only report on successful execution.  
+
+## Out-of-scope
+
+* Educational training or user assistance on how to create pipelines  
+* Updating backends so they support pipeline execution  
+* Updating pipelines so they are portable to environments  
+* Establishing standard data sets for portability (they are provided per pipeline)  
+* Adding 3rd party backends  
```

Comment:
Can you clarify why this is out of scope? The objectives state "The Portability Service implementation team as well as third parties can submit environments to be included in the Portability Service's test set" and one of the milestones is that "More than one backend are integrated" so it sounds like implementing the backends would be in scope. Perhaps there is something about the relationship between the portability service team and the third parties that could be stated here?

Reply:
Implementing backends is in scope but not for 3rd parties. We would implement the backends we specifically need for the project but enable others to add to the system (or "register" themselves if they were interested.

Reply:
Thanks Tim! My understanding is that implementing Cromwell support in the portability service is in scope because it is used for pipeline execution in the HCA. In addition, implementing at least one other backend is also in scope to demonstrate portability. But in general, the service would enable the developers of other backends to "register" themselves but it is out of scope for the portability team to add every possible backend. Does that sound right?